.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,25 @@
+name: Unit_test
+on: [push]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.allow-failures }}
+    strategy:
+      matrix:
+        ruby: ['3.3']
+        allow-failures: [false]
+        include:
+          - ruby: head
+            allow-failures: true
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Ruby ${{ matrix.ruby }}
+        uses: ruby/setup-ruby@v1
```

Comment:
I also think that in order to speed up the work of unit, we need to add caching of gems

```md
      - uses: ruby/setup-ruby@v1
        with:
          ruby-version: '3.3'
          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
```

https://github.com/ruby/setup-ruby

Reply:
The cache will be overwritten after making changes to the gem file